src/metabase/automagic_dashboards/populate.clj

```diff
@@ -43,9 +44,23 @@
         :location "/")
       (create-collection! "Automatically Generated Dashboards" "#509EE3" nil nil)))
 
-(def colors
+(defn- whitelabeled-colors
+  "Get color from application-colors or default color."
+  [default-colors-map]
+  (vec (for [[whitelabeled-color-name default-color] default-colors-map]
+         (get (public-settings/application-colors) whitelabeled-color-name default-color))))
```

Comment:
Since I'm changing from iterating a vector to iterating a map, I'm not sure if this is idiomatic, but hope it looks better @camsaul

Reply:
Some feedback:
- calling `(public-settings/application-colors)` repeatedly seems unnecessary. Just get the value once and then get values from it.
- maps are unordered. Getting a `vec` from iterating over it provides an undefined order. Perhaps you could use a vector of color names in a specified order? If the order is not important here, perhaps a docstring could help reassure. But the standard reaction when getting vec from a hashmap without specifying order is one of nervousness: the datastructure is almost entirely about order.

Reply:
@dpsutton 
> Some feedback:
> 
> * calling `(public-settings/application-colors)` repeatedly seems unnecessary. Just get the value once and then get values from it.

I still have limited knowledge on Clojure, could you point out how I could do that?

> * maps are unordered. Getting a `vec` from iterating over it provides an undefined order. Perhaps you could use a vector of color names in a specified order? If the order is not important here, perhaps a docstring could help reassure. But the standard reaction when getting vec from a hashmap without specifying order is one of nervousness: the datastructure is almost entirely about order.

Ahhh, I think the order matters based on the previous code. There's a version that would preserve the order https://github.com/metabase/metabase/pull/22458#discussion_r866204156. But the point of using a map is to really tell which color HEX maps to which color name. Is there a way to use a map and still maintain the color order?

Reply:
Something like the following:
- merge the defaults and the application settings so that a single map has the correct mapping from color type to color
- define an order that we select them in
- then, select them in that order
  - I use `(into [] ...)` because it is imperative that this is a vector since it is used as a function from index to color when using the hash in `map-to-colors` below. I like `into []` over `mapv` since it is easy to miss the `v` in `mapv` when the vector is critical.
- bonus, make a test that uses some temporary colors for the app and ensure they are selected

```clojure
(defn colors
  "A vector of colors useful for the application. Uses [[public-settings/application-colors]] for user choices."
  []
  (let [order  [:brand :accent1 :accent2 :accent3 :accent4 :accent5 :accent6 :accent7]
        merged (merge {:brand   "#509EE3"
                       :accent1 "#88BF4D"
                       :accent2 "#A989C5"
                       :accent3 "#EF8C8C"
                       :accent4 "#F9D45C"
                       :accent5 "#F2A86F"
                       :accent6 "#98D9D9"
                       :accent7 "#7172AD"}
                      (public-settings/application-colors))]
    (into [] (map merged) order)))

populate=> (colors)
["#509EE3"
 "#88BF4D"
 "#A989C5"
 "#EF8C8C"
 "#F9D45C"
 "#F2A86F"
 "#98D9D9"
 "#7172AD"]
populate=> (public-settings/application-colors)
{}
populate=> (public-settings/application-colors {:brand "blue" :accent3 "magenta"})
"{\"brand\":\"blue\",\"accent3\":\"magenta\"}"
populate=> (colors)
["blue"
 "#88BF4D"
 "#A989C5"
 "magenta"
 "#F9D45C"
 "#F2A86F"
 "#98D9D9"
 "#7172AD"]
```

Reply:
Thanks, @dpsutton I learned a lot. But still needs to wrap my head around the *transducers* 😆 .

Also, I didn't write any test as I think it's gonna take too much time. Because I have to figure out
- Where to put tests
- How to write tests
- How to run tests
Maybe that's for later.